flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPoolImpl.java

```diff
@@ -561,7 +561,11 @@ private void tryFulfillSlotRequestOrMakeAvailable(AllocatedSlot allocatedSlot) {
 			allocatedSlots.add(pendingRequest.getSlotRequestId(), allocatedSlot);
 			pendingRequest.getAllocatedSlotFuture().complete(allocatedSlot);
 
-			maybeRemapOrphanedAllocation(allocationIdOfRequest, allocatedSlot.getAllocationId());
+			// the allocation id can be null if the request was fulfilled by a slot directly offered
+			// by a reconnected TaskExecutor before the ResourceManager is connected
+			if (allocationIdOfRequest != null) {
```

Comment:
I think you could directly ask `pendingRequest.getAllocationId()`. That way you would not need to query `pendingRequests` again.

Reply:
Moreover, there is no invariant which states that `pendingRequests` contains the mapping. Hence, it would also make sense to directly ask `pendingRequest` here.

Reply:
Agreed. done.